### PR TITLE
Add support for @with_and_without_javascript cucumber tag

### PR DESCRIPTION
**Pull request by dwhenry**
This will allow us to run JS tests with javascript
enabled and disabled without having the test in the 
suite twice. This is based on the idea used at [MAS](https://github.com/moneyadviceservice/mortgage_calculator/blob/b6bcde64255305a81181d2ec5d784412eace5f31/features/support/cucumber_extensions.rb)

### CODE REVIEW

**Comment by andrewgarner** [comment]
What do you think @benlovell ?

File: spec/cucumber/with_and_without_javascript_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+          before do
+            allow(visitor).to receive(:feature).and_yield(visitor)
+            allow(visitor).to receive(:scenario).and_yield(visitor)
+            allow(visitor).to receive(:step).and_yield(visitor)
+          end
+
+          it 'Generates an @javascript and @no-javascript tagged scenario' do
+            expect(visitor).to receive(:scenario) do |scenario|
+              expect(scenario.tags).to eq([expected, expected_no_js])
```
Comment (benlovell) [inline spec/cucumber/with_and_without_javascript_spec.rb]: `contain_exactly` would be more concise here if you're matching the on the exact elements and ordering.

File: spec/cucumber/with_and_without_javascript_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            end.once
+            expect(visitor).to receive(:scenario) do |scenario|
+              expect(scenario.tags).to eq([expected, expected_js])
```
Comment (benlovell) [inline spec/cucumber/with_and_without_javascript_spec.rb]: Same here.

File: spec/cucumber/with_and_without_javascript_spec.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+            feature.describe_to(visitor)
+          end
+
+          context 'when RUN_JAVASCRIPT_TESTS_ONLY environment variable is set' do
+            it 'Generates only a @javascript tagged scenario' do
+              allow(ENV).to receive(:[]).with('RUN_JAVASCRIPT_TESTS_ONLY').and_return('true')
+
+              expect(visitor).to receive(:scenario) do |scenario|
+                expect(scenario.tags).to eq([expected, expected_js])
```
Comment (benlovell) [inline spec/cucumber/with_and_without_javascript_spec.rb]: Same here.

**Comment by benlovell** [comment]
I might be missing something blindingly obvious here, but for the longest time cucumber has supported multiple tags per feature/scenario. Could this work be negated by simply tagging features/scenarios using both the `@javascript` and `@no_javascript` where necessary?

**Comment by dwhenry** [comment]
Look at doing this with test runner one with javascript enabled and once without

**Comment by dwhenry** [comment]
Going to close this and look to use command line to run test suite multiple times, once with JS enable and once without rather than hack cucumber interbals.

**Pull request closed without merging**